### PR TITLE
Update autorest config

### PR DESCRIPTION
**Pull request by LuChen-Microsoft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/ChatThreadsImpl.java [hunk 1]
```diff
@@ -674,7 +674,7 @@ public Response<SendChatMessageResult> sendChatMessageWithResponse(
      * Gets a list of messages from a thread.
      *
      * @param chatThreadId The thread id of the message.
-     * @param maxPageSize The maximum number of messages to be returned per page. The limit can be found at https://docs.microsoft.com/en-us/azure/communication-services/concepts/service-limits#size-limits.
+     * @param maxPageSize The maximum number of messages to be returned per page.
      * @param startTime The earliest point in time to get messages up to. The timestamp should be in RFC3339 format:
```
Comment (LuChen-Microsoft) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/ChatThreadsImpl.java:678]: @jiminwen-msft did you change the description in the swagger JSON file?

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/models/SendTypingNotificationRequest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.communication.chat.implementation.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/** Request payload for typing notifications. */
+@Fluent
+public final class SendTypingNotificationRequest {
```
Comment (angiurgiu) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/models/SendTypingNotificationRequest.java:12]: We already have this class as TypingNotificationOptions

Comment (angiurgiu) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/models/SendTypingNotificationRequest.java:12]: This transformation from the Swagger/Readme should've renamed it, if it doesn't anymore something might've changed:

directive:
- from: swagger-document
  where: $.definitions
  transform: >
    if (!$.TypingNotificationOptions) {
      $.TypingNotificationOptions = $.SendTypingNotificationRequest;
      delete $.TypingNotificationRequest;
    }
- from: swagger-document
  where: $["paths"]["/chat/threads/{chatThreadId}/typing"].post.parameters[2]
  transform: >
    if ($.schema && $.schema.$ref && $.schema.$ref.endsWith("SendTypingNotificationRequest")) {
        const path = $.schema.$ref.replace(/[#].*$/, "#/definitions/TypingNotificationOptions");
        $.schema = { "$ref": path };
    }

**Review by angiurgiu — CHANGES_REQUESTED** [review]

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/models/SendTypingNotificationRequest.java (hunk 2 above)
Comment (LuChen-Microsoft) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/models/SendTypingNotificationRequest.java]: `delete $.TypingNotificationRequest;` 
The delete param name is incorrect here. Fixed it in the new iteration.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/ChatThreadsImpl.java (hunk 1 above)
Comment (jiminwen-msft) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/ChatThreadsImpl.java:678]: I changed json files in the spool gateway package. Not sure why it is not reflected while generating auto rest.

**Review by angiurgiu — APPROVED** [review]

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/ChatThreadsImpl.java (hunk 1 above)
Comment (LuChen-Microsoft) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/implementation/ChatThreadsImpl.java:678]: You need to update here https://github.com/Azure/azure-rest-api-specs/tree/main/specification/communication/data-plane/Chat

**Pull request merged**